src/ex2/ItemManager.js

```diff
@@ -0,0 +1,170 @@
+import UIManager from "./UIManager.js";
+import Item from "./ItemClass.js";
+import PokemonClient from "./PokemonClient.js";
+
+class ItemManager {
+  constructor() {
+    this.itemIdCounter = 2000;
+    this.pokemonClient = new PokemonClient();
+    this.uiManager = new UIManager();
+    this.fethcedPokemons = [];
+    this.pokemonErrorBatch = [];
+    this.itemList = [];
+    this.allPokemonsJson = null;
+    this.clearAllBtn = document.getElementById("clearAllBtnId");
+    // this.trashButton = null;
+    this.clearAllBtn.addEventListener("click", () => {
+      this.handleClearAllTodos();
+    });
+  }
+
+  /**
+   *
+   * @param {string} inputValue
+   * @returns
+   */
+  parseInputValue(inputValue) {
+    inputValue = inputValue.trim();
+    if (!inputValue) {
+      return alert("Error, the Todo input is empty");
+    }
+
+    let promises = [];
```

Comment:
since you are not changing the reference to "promises", according to best practices it should be const (you add to the array but don't change the array to a new array).